src/components/LogoutModal.js

```diff
@@ -0,0 +1,143 @@
+import { deleteADocument, getADocReferenceFromCollection, updateADocument, getAllDocuments } from '../../firebase/helpers/database.firebase';
+
+const LogoutModal = ({ modalVisible, setModalVisible }) => {
+  const { state: { guests, accountType, sessionId, nickname }, dispatch } = useContext(BiteShareContext);
+  const navigation = useNavigation();
+  const styles = StyleSheet.create({
+    centeredView: {
+      flex: 1,
+      justifyContent: 'center',
+      alignItems: 'center',
+      marginTop: 22,
+    },
+    modalView: {
+      margin: 20,
+      backgroundColor: colors.brand.login,
+      borderRadius: 20,
+      padding: 35,
+      alignItems: 'center',
+      shadowColor: '#000',
+      shadowOffset: {
+        width: 0,
+        height: 2
+      },
+      shadowOpacity: 0.25,
+      shadowRadius: 4,
+      elevation: 5
+    },
+    button: {
+      borderRadius: 20,
+      padding: 10,
+      width: 100,
+      elevation: 2
+    },
+    buttonLogout: {
+      backgroundColor: colors.brand.kazan,
+      marginBottom: 10
+    },
+    logout: {
+      color: 'white'
+    },
+    buttonClose: {
+      backgroundColor: colors.brand.beach,
+    },
+    textStyle: {
+      color: 'black',
+      fontWeight: 'bold',
+      textAlign: 'center'
+    },
+    modalText: {
+      fontSize: 16,
+      marginBottom: 15,
+      textAlign: 'center'
+    }
+  });
+
+  const logout = async () => {
+    console.log('logout Data', sessionId, accountType);
+    if (sessionId && accountType === 'HOST') {
+      try {
+        const qResult = await getADocReferenceFromCollection(`transactions/${sessionId}/attendees`, 'name', '==', nickname);
+        qResult.forEach((doc) => {
+          // console.log('DOCSSS', doc.data());
+          updateADocument(`transactions/${sessionId}/attendees`, doc.id, {
+            isSessionActive: false
+          });
+        });
+        const getAttendees = await getAllDocuments(`transactions/${sessionId}/attendees`);
+        getAttendees.forEach(async (attendee) => {
+          await updateADocument(`transactions/${sessionId}/attendees`, attendee.id, {
+            joinRequest: 'denied'
+          });
+        });
+        await deleteADocument('transactions', sessionId);
+      } catch (err) {
+        console.log('Error denying the host: ', err);
+      }
+    } else if (sessionId && accountType === 'GUEST') {
+      try {
+        const qResult = await getADocReferenceFromCollection(`transactions/${sessionId}/attendees`, 'name', '==', nickname);
+        qResult.forEach((doc) => {
+          console.log('Guest has left');
+          updateADocument(`transactions/${sessionId}/attendees`, doc.id, {
+            joinRequest: 'denied',
+          });
+        });
+      } catch (err) {
+        console.log('Error denying the guest: ', err);
+      }
+    }
+    signOutUser()
+      .then(() => {
+        dispatch({ type: 'SET_CLEAR_CONTEXT' });
```

Comment:
lol do we need to clear more context variables than this reducer currently covers? like `accountHolderName`, `nickname` etc. If so, then we can add another reducer function to cover every context that needs to cleared on logout

Reply:
Yeah, I'm glad you brought this up, because I wasn't sure if it covers all necessary context haha. I assumed the way you built it that it would cover everything I needed lol :)